tests/suites/test_suite_x509parse.function

```diff
@@ -1250,6 +1250,86 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_FS_IO:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_CERTS_C:MBEDTLS_SHA1_C:MBEDTLS_RSA_C */
+void mbedtls_x509_crt_parse_subjectkeyid( char * crt_path, char *subjectKeyId, int subjectKeyIdLength, int retVal )
+{
+    mbedtls_x509_crt crt;
+    int i = 0;
+    int result = 0;
+
+    mbedtls_x509_crt_init( &crt );
+
+    TEST_ASSERT( mbedtls_x509_crt_parse_file( &crt, crt_path ) == retVal );
+
+    if(retVal != 0)
+    {
+        while(i < subjectKeyIdLength)
+        {
+            result |= crt.subject_key_id.p[i] != subjectKeyId[i*2];
+            result |= crt.subject_key_id.p[i+1] != subjectKeyId[i*2+1];
+            i++;
+        }
+
+        TEST_ASSERT(result == 0);
+    }
```

Comment:
Added the validation. The value of the keyID and authorityKeyId.KeyID validation was removed. It was not working, but I did not realize this because of the wrong condition (ref_ret != 0). I checked the other test and this kind of serial numbers are not checked anywhere and I was not able to implement it. If I pass a character string through test argument (e.g. "\xd4\xa5") it will not be converted to characters. I do not know why this happens, but I think checking the success of the parsing, the tag and the length is good enough (as in the other tests)